include/xtensor/utils/xutils.hpp

```diff
@@ -283,29 +283,31 @@ namespace xt
      * apply implementation *
      ************************/
 
-    namespace detail
-    {
-        template <class R, class F, std::size_t I, class... S>
-        R apply_one(F&& func, const std::tuple<S...>& s) NOEXCEPT(noexcept(func(std::get<I>(s))))
-        {
-            return static_cast<R>(func(std::get<I>(s)));
-        }
-
-        template <class R, class F, std::size_t... I, class... S>
-        R apply(std::size_t index, F&& func, std::index_sequence<I...> /*seq*/, const std::tuple<S...>& s)
-            NOEXCEPT(noexcept(func(std::get<0>(s))))
-        {
-            using FT = std::add_pointer_t<R(F&&, const std::tuple<S...>&)>;
-            static const std::array<FT, sizeof...(I)> ar = {{&apply_one<R, F, I, S...>...}};
-            return ar[index](std::forward<F>(func), s);
-        }
-    }
-
     template <class R, class F, class... S>
     inline R apply(std::size_t index, F&& func, const std::tuple<S...>& s)
-        NOEXCEPT(noexcept(func(std::get<0>(s))))
     {
-        return detail::apply<R>(index, std::forward<F>(func), std::make_index_sequence<sizeof...(S)>(), s);
+        return std::apply(
+            [&](const S&... args) -> R
+            {
+                return [&](const auto&... t) -> R
```

Comment:
Why do we need an additional lambda here? Can't we have the implementation directly in the lambda passed to ` std::apply`  ?

Reply:
Oops. Yeah... My bad.